### ISSUE TITLE
Error with Crystal nightly

### ISSUE THREAD

**Issue by jwoertink** [luckyframework/avram#1025]
This just recently popped up in the last few days.

![image](https://github.com/luckyframework/avram/assets/2391/d2d2614e-01e7-4a86-89af-447dae5dd4d7)

Maybe related? https://github.com/luckyframework/lucky/issues/1872

**Comment by jwoertink** [comment]
More coming up now 

![image](https://github.com/luckyframework/avram/assets/2391/a2ed4dc3-3076-476a-97a4-fc8b0fde8ff3)

**Comment by davidepaolotua** [comment]
I kept getting errors in tests...

TL;DR: with the changes in the nightly build - optional/nillable settings in habitat are now forbidden

Just my two cents... most errors I've been seeing come directly from Habitat.
It seems that for whatever reason, when you check for raise_if_missing_settings, in the nightly build the check is way more stringent now. E.g:
Before, it was letting it pass if the setting was optional and you didn't provide it (aka: assuming it was there and nil)
Now, it fails. Aka: you have to *always* provide it - and setting it to null does not work (I didn't understand why though, but I didn't have the time to investigate it further)

What to do with that? Dunno :/ I doubt that even providing a default value would help in that case...

**Comment by jwoertink** [comment]
Yeah, I figured something might be up with that. I was trying to [test a patch](https://github.com/crystal-lang/crystal/issues/13193#issuecomment-2103475708) in Crystal and couldn't even get that to run because I was getting weird errors. 

I'll see if I can dig in more to find out. Will probably need to do a git bisect since v1.12.1 to see what changed.
